jams/jamongo.py

```diff
@@ -0,0 +1,171 @@
+"""
+JAMS Mongo API
+==============
+
+This library provides an interface for interfacing with a JAMS Mongo
+instance.
+
+"""
+
+import pymongo
+
+
+def convert_annotation_list(annotations_array):
+        """Take a "Jams-document" style annotations array,
+        and convert it into a mongo-able annotationsl list.
+
+        Parameters
+        ----------
+        annotations_list : jams.AnnotationArray
+
+        Returns
+        -------
+        converted_annotations : list
+        """
+        return []
+
+
+class JamsMongo(object):
+    """Model class for interfacing with JAMS Mongo Collection."""
+
+    def __init__(self, connection_str=None,
+                 client=None,
+                 database_name="jams"):
+        """
+        Pass the parameters necessary to create a MongoClient connection to
+        mongodb.
+
+        If no parameters are provided, just tries to connect to your
+        local mongod.
+
+        Parameters
+        ----------
+        connection_str : str or None
+            Mongo connection string.
+
+        client : pymongo.mongo_client.MongoClient
+            A MongoClient reference. Use this if you want to pass the client
+            instead of having JamsMongo create the client for you from the
+            connection_str.
+
+            (This is useful for unit tests / mongomock.)
+
+        database_name : str
+            The name of the database to use on your client.
+        """
+        self.connection_str = connection_str
+        self.client = client
+        self.database_name = database_name
+        self._db = None
+
+    def __enter__(self):
+        """Set up your mongo collection, and return the object to access it."""
+        self._connect_to_mongo()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        pass
+
+    @property
+    def db(self):
+        return self._db
+
+    @property
```

Comment:
We have "annotations" and "audio" here... should there be somewhere that explicitly defines these collections?  And should these getters be attributes, or would it make more sense to provide a dict interface in this object where the key maps to a collection in `_db`?